src/Compilers/Core/Portable/PEWriter/MetadataWriter.PortablePdb.cs

```diff
@@ -831,5 +832,46 @@ private void EmbedSourceLink(Stream stream)
                 kind: _debugMetadataOpt.GetOrAddGuid(PortableCustomDebugInfoKinds.SourceLink),
                 value: _debugMetadataOpt.GetOrAddBlob(bytes));
         }
+
+        private void EmbedCompilerFlags(CommonPEModuleBuilder module)
+        {
+            byte[] bytes = new byte[0];
+
+            // TODO: Where to get compiler version? 
+
+            // TODO: Where to get source encoding
+
+            _debugMetadataOpt.AddCustomDebugInformation(
+                parent: EntityHandle.ModuleDefinition,
+                kind: _debugMetadataOpt.GetOrAddGuid(PortableCustomDebugInfoKinds.DeterministicCompilerFlags),
+                value: _debugMetadataOpt.GetOrAddBlob(bytes));
+        }
+
+        private void EmbedMetadataReferenceInformation(CommonPEModuleBuilder module)
+        {
+            var bytes = new List<byte>();
+
+            // TODO: Get metadata reference information
+            // File name: Foo.exe
+            // COFF header Timestamp field (4 bytes): 0x542d5742
+            // COFF header SizeOfImage field (4 bytes): 0x32000000
+            // MVID (Guid, 24 bytes): 0x24a44d8218894463807674caf3b1c19a
+            // Output: Foo.exe\0542d5742320000000024a44d8218894463807674caf3b1c19a
+            foreach (var metadataReference in module.CommonCompilation.ExternalReferences)
+            {
+                if (metadataReference.Display is object && !IsTooLongInternal(metadataReference.Display, PdbLengthLimit))
+                {
+                    bytes.AddRange(Encoding.UTF8.GetBytes(metadataReference.Display));
```

Comment:
>metadataReference.Display [](start = 58, length = 25)

We need to emit the file name with extension. `Display` may not be that - it's potentially an arbitrary string that is displayed in error messages.

I think we need to distinguish between PE references and Compilation references here. For our scenario, we are only interested in PE reference. Compilation references are only created by the IDE. I think the best thing to do here is to skip them. That is only emit references that are `PortableExecutableReference` and then we can use the `FilePath` property (also skip references whose `FilePath` is null). Since the path is a full path we would want to call `PathUtilities.GetFileName` on it.